### PR TITLE
Category component

### LINKED ISSUE #11: Create Category component
We want to create the main `Category` component that contains all of its subcategories and their entries. It also contains a budget progress component. 

Entry

The data strategy for a category is very similar to `SubCategory`. Much like a SubCategory has an array of Entry components, the `Category` component will have its own array of `SubCategory` data, each on in turn containing `Entries`. This is more or less a one-level-up abstraction of what we have already, your challenge is to try to figure out the final data shape on your own. Provided is the sample of Category meta fields:

```javascript
const data = {
  id: 'entertainment,
  title: 'Entertainment,
  description: 'sample description',
  budget: 500,
  subCategories: [
    // for you to determine what goes in here, think back to how we handled SubCategory and Entry data
  ]
}
```

### Should

**Header**
* [x] Display a placeholder 'circle' (later to be an icon) as the leftmost element
* [x] Display the title and description, stacked, to the left of the circle placeholder
* [x] Display the `CategoryProgress` component as the rightmost component
* [x] Calculate the total category amount, as determined by a sum of all of it's entries across all of it's sub-categories. Don't overthink this, its fine to do a top level add reduce, and have a `SubCategory` do its reduce too, reduce is very efficient.
  * [x] Pass the calculated value to the `CategoryProgress` component, along with the `budget` meta field so it can render itself accordingly.

**Body**
* Render a list of `SubCategories` below the header

### Styles

* circle color: `#46237A` (look into border radius for creating the shape)
* circle diamater: `48px`
* Category title: `#4A4A4A` / `22px` / `500`
* Category description: `#9B9B9B` / `12px`

> NOTE: SubCategories **DO NOT** need to be aligned left to match up with Category title. The design has it this way, Im changing the requirement here.

### Depends On
* #6

### PR DESCRIPTION
**Pull request by megdadkins7**
resolves #11 created category component

### CODE REVIEW

File: src/components/Category.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+const categoryTotal = data.subCategories.reduce(
+  (nextTotal, subCategoryBudget) => nextTotal + subCategoryBudget.total,
+  0
+)
+
+export default function Category() {
+  return (
+    <StyledCategory>
+      <div className="CategoryWrapper">
+        <div className="CategoryLeftChild">
```
Comment (pixx3l) [inline src/components/Category.js]: **OPTIONAL** in the future, we probably want to use more generic class names like `ShiftLeft` or better yet, abstract the group into it's own `styled-component` file so they can be reused in multiple places, but it's totally fine for now

File: src/components/SubCategory.js [hunk 2]
```diff
@@ -24,36 +23,40 @@ const StyledSubCategory = styled.div`
...
+        <span className="SubCategory-total">
+          {numToCurrency(subCategoryTotal)}
+        </span>
       </div>
-      <ul className='SubCategory-entries'>
-        {subCategoryEntries.map(entryItem => 
-           <Entry {...entryItem} />
-        )}
+      <ul className="SubCategory-entries">
+        {subCategoryEntries.map(entryItem => (
```
Comment (pixx3l) [inline src/components/SubCategory.js]: es-lint should be yelling at you here cause you don't have a key, ping me when you're working and we'll go over what it is

File: src/components/SubCategory.js [hunk 3]
```diff
@@ -24,36 +23,40 @@ const StyledSubCategory = styled.div`
...
-  }
-      
-const subCategoryEntries = [
-  {title: "Mistborn", amount: 9.99},
-  {title: "Name of the Wind", amount: 24.95},
-  {title: "The Dark Knight", amount: 8.99},
-  {title: "The Marvelous Mrs. Maisel", amount: 4.92},
-  {title: "Mumford & Sons Vinyl", amount: 15.65}
-];
+  id: 'unique',
```
Comment (pixx3l) [inline src/components/SubCategory.js]: this was meant to be a placeholder, the point was that the `id` should be a unique title

**Comment by megdadkins7** [comment]
resolves #11 added unique id and keys, will refactor later with a little more understanding on how to do so effectively.

**Pull request merged**